pkg/autoscaler/metrics/http_scrape_client_test.go

```diff
@@ -250,3 +279,20 @@ func newTestHTTPClient(response *http.Response, err error) *http.Client {
 		},
 	}
 }
+
+func BenchmarkUnmarshalling(b *testing.B) {
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		b.StopTimer()
+		hClient := newTestHTTPClient(makeProtoResponseForBenchmarking(http.StatusOK, Stat{}, network.ProtoAcceptContent), nil)
+
+		scrapeClient, err := newHTTPScrapeClient(hClient)
+		if err != nil {
+			b.Fatal(err)
+		}
```

Comment:
I would like to have different http clients based on the response because the response is randomized and I cannot set the response otherwise.

Reply:
We could use `ReadAtLeast` but it is the same code:
```
func ReadAtLeast(r Reader, buf []byte, min int) (n int, err error) {
	if len(buf) < min {
		return 0, ErrShortBuffer
	}
	for n < min && err == nil {
		var nn int
		nn, err = r.Read(buf[n:])
		n += nn
	}
	if n >= min {
		err = nil
	} else if n > 0 && err == EOF {
		err = ErrUnexpectedEOF
	}
	return
}
```
I just wanted to get rid of the redundant checks because i already know the size of the buffer.

Reply:
Does the response need to be randomized? If we'd consider gzipping, it might make a difference but does it in our case? Seems like just using a "real" podname would be sufficient.

Reply:
@markusthoemmes afaik the podname size makes a difference when you calculate the marshaled size of the data. This seems more realistic to me than assuming all calls return the same data.

Reply:
It does make the tests non-deterministic though. Personally, I'd just run the worst case in the benchmark, or run a best (shortest, all 0s), medium (mid pod name, some values) and worst case scenario respectively to test them all reliably without introducing a jitter into the test.